File: TY.Services.Bank/TY.Services.Bank/Models/Account/Account.cs
using System;
using Newtonsoft.Json;

namespace TY.Services.Bank.Models.Account
{
    public class Account : IBaseEntity
    {
        [JsonProperty("_id")]
        public string Id { get; set; }

        [JsonProperty("isActive")]
        public bool IsActive { get; set; }

        [JsonProperty("createdAt")]
        public DateTime? CreatedAt { get; set; }

        [JsonProperty("updatedAt")]
        public DateTime? UpdatedAt { get; set; }

        [JsonProperty("firstName")]
        public string FirstName { get; set; }

        [JsonProperty("lastName")]
        public string LastName { get; set; }

        [JsonProperty("address")]
        public string Address { get; set; }

        [JsonProperty("email")]
        public string Email { get; set; }

        [JsonProperty("phoneNumber")]
        public string PhoneNumber { get; set; }

        [JsonProperty("age")]
        public int Age { get; set; }

        [JsonProperty("salt")]
        public string Salt { get; set; }

        [JsonProperty("password")]
        public string Password { get; set; }

        [JsonProperty("balance")]
        public decimal Balance { get; set; }
    }
}

File: TY.Persistent.Bank/models/Account.js
import { set, Schema, model } from 'mongoose'
import bcrypt from 'bcrypt'
set('debug', true)

const AccountSchema = new Schema({
    firstName: {
        type: String,
        required: true,
        min: 3,
        max: 50
    },
    lastName: {
        type: String,
        required: true,
        min: 3,
        max: 100,
        trim: true
    },
    address: {
        type: String,
        trim: true,
        max: 255
    },
    email: {
        unique: true,
        type: String,
        trim: true,
        index: true
    },
    phoneNumber: {
        type: String,
        trim: true,
        default: null
    },
    age: {
        type: Number,
        default: 0,
        required: true
    },
    salt: {
        type: String,
        default: null,
        trim: true
    },
    password: {
        type: String,
        min: 6,
        required: true,
        trim: true
    },
    balance: {
        type: Number,
        required: true,
        default: 0
    },
    isActive: {
        type: Boolean,
        default: true
    }
}, {
    timestamps: true,
    toJSON: {
        virtuals: true
    }
})

AccountSchema.set('toJSON', {
    transform: function(doc, ret, opt) {
        delete ret['password']
        return ret
    }
})

model('Account', AccountSchema)